src/SourceCode.Clay.OpenApi/Example.cs

```diff
@@ -16,72 +18,106 @@ namespace SourceCode.Clay.OpenApi
     /// The <c>Value</c> property is implementation-specific. If it is required, create a new type
     /// that inherits from this one.
     /// </remarks>
-    public class Example : IEquatable<Example>
+    public sealed class Example : IEquatable<Example>
     {
+        #region Fields
+
+        private readonly JsonObject _json;
+
+        #endregion
+
         #region Properties
 
         /// <summary>
         /// Gets the short description for the example.
         /// </summary>
-        public string Summary { get; }
+        public string Summary
+        {
+            get
+            {
+                if (_json.TryGetValue(OpenApiSerializer.PropertyConstants.Summary, out var jv))
+                    return jv;
+
+                return null;
+            }
+        }
 
         /// <summary>
         /// Gets the long description for the example.
         /// </summary>
         /// <remarks>CommonMark syntax MAY be used for rich text representation.</remarks>
-        public string Description { get; }
+        public string Description
+        {
+            get
+            {
+                if (_json.TryGetValue(OpenApiSerializer.PropertyConstants.Description, out var jv))
+                    return jv;
+
+                return null;
+            }
+        }
 
         /// <summary>
         /// Gets the URL that points to the literal example.
         /// </summary>
-        public Uri ExternalValue { get; }
+        public Uri ExternalValue
+        {
+            get
+            {
+                if (_json.TryGetValue(OpenApiSerializer.PropertyConstants.Url, out var jv))
+                    return new Uri(jv);
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Gets custom properties.
+        /// </summary>
+        public JsonObject VendorExtension { get; }
```

Comment:
This example uses `JsonObject` pattern. It is much simpler than the previous pattern, but it remains a mutable property.